### PR TITLE
feature/cplp-1723/idp user list

### PR DESCRIPTION
**Pull request by oyo**
PR for IDP management
Note: there are 23 code smells about arrow functions as props. We are ignoring these for now because we think this is an issue that has to be fixed in the React framework and this is already in discussion there.

### CODE REVIEW

File: cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx [hunk 1]
```diff
@@ -65,46 +87,103 @@ export default function IDPListItem({ idp }: { idp: IdentityProvider }) {
...
-          ) : (
-            <IconButton color="secondary" onClick={doEnableToggle}>
-              <ToggleOffIcon />
-            </IconButton>
-          )}
+  const renderButtons = () => {
+    return (
+      <>
+        {/*
+        <IconButton
```
Comment (nidhigarg-bmw) [inline cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx:94]: Remove this commented code if not needed

File: cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx [hunk 2]
```diff
@@ -65,46 +87,103 @@ export default function IDPListItem({ idp }: { idp: IdentityProvider }) {
...
+          </IconButton>
+        )}
+      </>
+    )
+  }
+
+  const renderMenu = () => {
+    return (
+      <DropdownMenu buttonText={t('action.actions')}>
+        {/*<MenuItem title={t('action.details')} onClick={toggle} />*/}
```
Comment (nidhigarg-bmw) [inline cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx:153]: Here also

File: cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx [hunk 3]
```diff
@@ -65,46 +87,103 @@ export default function IDPListItem({ idp }: { idp: IdentityProvider }) {
...
+        <span className="alias">{idp.alias}</span>
+        <span className="state">
+          <Checkbox disabled={true} checked={idp.enabled} />
+        </span>
+        <span className={`action ${buttons ? 'buttons' : 'menu'}`}>
+          {buttons ? renderButtons() : renderMenu()}
         </span>
       </div>
-      {open && (
+      {/*open && (
```
Comment (nidhigarg-bmw) [inline cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx:182]: Here as well

File: cx-portal/src/components/pages/Test/IDPTest/index.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+export default function IDPTest() {
+  const update = useSelector(updateIDPSelector)
+  const userdata = useFetchOwnUserDetailsQuery().data
+  const idpdata = useFetchIDPListQuery(update).data
+
+  return (
+    <section className="idptest">
+      {'userdata'}
+      <pre>{JSON.stringify(userdata, null, 2)}</pre>
```
Comment (nidhigarg-bmw) [inline cx-portal/src/components/pages/Test/IDPTest/index.tsx:51]: Here, are we displaying raw data ?

File: cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx (hunk 1 above)
Comment (oyo) [reply cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx:94]: done

File: cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx (hunk 3 above)
Comment (oyo) [reply cx-portal/src/components/pages/IDPManagement/IDPListItem.tsx:182]: done

File: cx-portal/src/components/pages/Test/IDPTest/index.tsx (hunk 4 above)
Comment (oyo) [reply cx-portal/src/components/pages/Test/IDPTest/index.tsx:51]: removed

**Review by nidhigarg-bmw — APPROVED** [review]
looks good now

**Pull request merged**